defaults/policies/advice/otel.rego

```diff
@@ -0,0 +1,97 @@
+package advice
+
+import rego.v1
+
+# Use pre-computed sets from jq
+attributes_set := data.attributes_set
+
+templates_set := data.templates_set
+
+namespaces_to_check_set := data.namespaces_to_check_set
+
+name_regex := "^[a-z][a-z0-9]*([._][a-z0-9]+)*$"
+
+derive_namespaces(name) := [
+concat(".", array.slice(parts, 0, i)) |
+	parts := split(name, ".")
+	count(parts) > 1 # Only derive namespaces if there are at least 2 parts
+
+	# Stop at count(parts)-1 to exclude the full attribute name
+	some i in numbers.range(1, count(parts) - 1)
+]
+
+# checks attribute has a namespace
+deny contains make_advice(key, advisory, value, message) if {
+	not contains(input.name, ".")
+	key := "missing_namespace"
+	advisory := "improvement"
+	value := input.name
+	message := "Does not have a namespace"
+}
+
+# checks attribute name format
+deny contains make_advice(key, advisory, value, message) if {
```

Comment:
This is also something we do not require non-semconv attributes to have.

We should definitely use this for Semantic conventions, but don't think we need to force everyone to do it.  E.g. think "prometheus metrics"

Reply:
We could default to no Rego and then allow the `--advice-policies` to accept a path or the string `otel`. We could even have `prometheus` as an embedded policy too if there's a "standard.